### PR TITLE
Load reporting service documentation 

### PR DESCRIPTION
**Pull request by kathan24**
For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/master/PULL_REQUESTS.md)

Commit Message: Load Reporting Service documentation 
Additional Description:
Risk Level: Low
Testing:
Docs Changes: Yes, for Load Reporting Service
Release Notes:
[Optional Fixes #Issue]: [9651](https://github.com/envoyproxy/envoy/issues/9651)
[Optional Deprecated:]

### CODE REVIEW

File: api/envoy/config/endpoint/v3/load_report.proto [hunk 1]
```diff
@@ -17,11 +17,11 @@ option java_outer_classname = "LoadReportProto";
 option java_multiple_files = true;
 option (udpa.annotations.file_status).package_version_status = ACTIVE;
 
-// These are stats Envoy reports to GLB every so often. Report frequency is
-// defined by
+// [#protodoc-title: Load Report]
+
+// These are stats Envoy reports to GLB every so often. Report frequency is defined by
```
Comment (wgallagher) [inline api/envoy/config/endpoint/v3/load_report.proto:22]: I would reword this as 'These are stats Envoy reports to GLB at a frequency defined by...'

File: api/envoy/service/load_stats/v3/lrs.proto [hunk 2]
```diff
@@ -17,7 +17,12 @@ option java_multiple_files = true;
 option java_generic_services = true;
 option (udpa.annotations.file_status).package_version_status = ACTIVE;
 
-// [#protodoc-title: Load reporting service]
+// [#protodoc-title: Load reporting service (LRS)]
+
+// Load Reporting Service is an Envoy API to emit load reports. Envoy will initiate a bi-directional stream with a management server.
+// Upon connecting, the management server can send a LoadStatsResponse to a node it is interested in getting the load reports.
```
Comment (wgallagher) [inline api/envoy/service/load_stats/v3/lrs.proto:23]: typo: 'getting the load reports *for*'

File: api/envoy/config/endpoint/v3/load_report.proto [hunk 3]
```diff
@@ -17,11 +17,11 @@ option java_outer_classname = "LoadReportProto";
 option java_multiple_files = true;
 option (udpa.annotations.file_status).package_version_status = ACTIVE;
 
-// These are stats Envoy reports to GLB every so often. Report frequency is
-// defined by
+// [#protodoc-title: Load Report]
+
+// These are stats Envoy reports to GLB at a frequency defined by
```
Comment (htuch) [inline api/envoy/config/endpoint/v3/load_report.proto:22]: s/GLB/the management server/

File: api/envoy/service/load_stats/v3/lrs.proto [hunk 4]
```diff
@@ -17,7 +17,12 @@ option java_multiple_files = true;
 option java_generic_services = true;
 option (udpa.annotations.file_status).package_version_status = ACTIVE;
 
-// [#protodoc-title: Load reporting service]
+// [#protodoc-title: Load reporting service (LRS)]
+
+// Load Reporting Service is an Envoy API to emit load reports. Envoy will initiate a bi-directional stream with a management server.
+// Upon connecting, the management server can send a LoadStatsResponse to a node it is interested in getting the load reports for.
```
Comment (htuch) [inline api/envoy/service/load_stats/v3/lrs.proto:23]: Maybe link with `:ref:` to the corresponding messages using the `envoy_v3_api_msg` anchors.

Comment (mattklein123) [reply api/envoy/service/load_stats/v3/lrs.proto:23]: Please also line wrap around 100 col if possible, thanks.

File: api/envoy/service/load_stats/v3/lrs.proto [hunk 5]
```diff
@@ -17,7 +17,16 @@ option java_multiple_files = true;
...
+// [#protodoc-title: Load reporting service (LRS)]
+
+// Load Reporting Service is an Envoy API to emit load reports. Envoy will initiate a bi-directional
+// stream with a management server. Upon connecting, the management server can send a
+// :ref:`LoadStatsResponse <envoy_api_msg_service.load_stats.v3.LoadStatsResponse>` to a node it is
+// interested in getting the load reports for. Envoy in this node will start sending
+// :ref:`LoadStatsRequest <envoy_api_msg_service.load_stats.v3.LoadStatsRequest>`. This is done periodically
+// based on the :ref:`load reporting interval <envoy_api_field_service.load_stats.v3.LoadStatsResponse.load_reporting_interval>`
+// For details, take a look at the `sandbox example
+// <https://www.envoyproxy.io/docs/envoy/latest/start/sandboxes/load_reporting_service.html>`_.
```
Comment (mattklein123) [inline api/envoy/service/load_stats/v3/lrs.proto:29]: please ref link directly to the RST page vs. use a hard coded link.

File: docs/root/intro/arch_overview/upstream/load_reporting_service.rst [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+The Load Reporting Service provides a mechanism by which Envoy can emit Load Reports to a management
+server at a regular cadence. Below is the envoy config to enable Load reporting service.
+
+This will initiate a bi-directional stream with a management server. Upon connecting, the management
+server can send a :ref:`LoadStatsResponse <envoy_v3_api_msg_service.load_stats.v3.LoadStatsResponse>`
+to a node it is interested in getting the load reports for. Envoy in this node will start sending
+:ref:`LoadStatsRequest <envoy_v3_api_msg_service.load_stats.v3.LoadStatsRequest>`. This is done periodically
+based on the :ref:`load reporting interval <envoy_v3_api_field_service.load_stats.v3.LoadStatsResponse.load_reporting_interval>`
+
+For details, take a look at the `sandbox example <https://www.envoyproxy.io/docs/envoy/latest/start/sandboxes/load_reporting_service.html>`_.
```
Comment (mattklein123) [inline docs/root/intro/arch_overview/upstream/load_reporting_service.rst:15]: ref link

File: docs/root/intro/arch_overview/upstream/load_reporting_service.rst [hunk 7]
```diff
@@ -0,0 +1,39 @@
+.. _arch_overview_load_reporting_service:
+
+Load Reporting Service (LRS)
+============================
+
+The Load Reporting Service provides a mechanism by which Envoy can emit Load Reports to a management
+server at a regular cadence. Below is the envoy config to enable Load reporting service.
```
Comment (mattklein123) [inline docs/root/intro/arch_overview/upstream/load_reporting_service.rst:7]: nit: consistent capitalization around "Load Reporting"

File: docs/root/intro/arch_overview/upstream/load_reporting_service.rst [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+This will initiate a bi-directional stream with a management server. Upon connecting, the management
+server can send a :ref:`LoadStatsResponse <envoy_v3_api_msg_service.load_stats.v3.LoadStatsResponse>`
+to a node it is interested in getting the load reports for. Envoy in this node will start sending
+:ref:`LoadStatsRequest <envoy_v3_api_msg_service.load_stats.v3.LoadStatsRequest>`. This is done periodically
+based on the :ref:`load reporting interval <envoy_v3_api_field_service.load_stats.v3.LoadStatsResponse.load_reporting_interval>`
+
+For details, take a look at the `sandbox example <https://www.envoyproxy.io/docs/envoy/latest/start/sandboxes/load_reporting_service.html>`_.
+
+.. code-block:: yaml
```
Comment (mattklein123) [inline docs/root/intro/arch_overview/upstream/load_reporting_service.rst:17]: This can get out of date. I would recommend just `:repo:` linking to the config file in the example.

**Comment by mattklein123** [comment]
Please fix format.

/wait
